pkg/target/cluster.go

```diff
@@ -80,6 +100,53 @@ func (cl *Cluster) Send(cwg *sync.WaitGroup, finish chan struct{}) {
 	}()
 }
 
+func (cl *Cluster) removeAvailableHost(host *Host) {
+	for i, h := range cl.AvailableHosts {
+		if h.Name == host.Name {
+			cl.Hm.Lock()
+			defer cl.Hm.Unlock()
+			length := len(cl.AvailableHosts)
+
+			for j := i; j < length-2; j++ {
+				cl.AvailableHosts[i] = cl.AvailableHosts[i+1]
+			}
+			cl.AvailableHosts = cl.AvailableHosts[:length-1]
+
+			break
+		}
+	}
+}
+
+func (cl *Cluster) addAvailableHost(host *Host) {
+	cl.Hm.Lock()
+	defer cl.Hm.Unlock()
+	for _, h := range cl.AvailableHosts {
+		if h.Name == host.Name && host.Port == h.Port {
+			return
+		}
+	}
+	cl.AvailableHosts = append(cl.AvailableHosts, host)
+}
+
+func (cl *Cluster) keepAvailableHostsUpdated() {
+	for {
+		h := <-cl.UpdateHostHealthStatus
+		if cl.Type != conf.LB {
+			continue
+		}
+		cl.updateHostAvailability(*h)
+	}
+}
+
+func (cl *Cluster) updateHostAvailability(h HostStatus) {
+	defer close(h.sigCh)
+	if h.Status == false {
```

Comment:
We might want to track hosts flapping state.
Might be prometheus counter with amount of state changes per destination?

Reply:
This is added as well.